router/src/lib.rs

```diff
@@ -906,8 +906,18 @@ impl ChatRequest {
         let (inputs, grammar, using_tools) = prepare_chat_input(
             infer,
             response_format,
-            tools,
-            tool_choice,
+            tools.clone(),
+            // unwrap or default (use "auto" if tools are present, and "none" if not)
+            tool_choice.map_or_else(
```

Comment:
That should be already done in the deserialization, there are about 3 different function calls to `preprare_chat_input`.